tools/ping_enumerator.py

```diff
@@ -3,76 +3,145 @@
+        follows target to real device an links origin to it
+        origin => target
+        Returns True if sucessful
+        """
         try:
-            device_id = new_ping.get_device_id()
-            firmware_version = new_ping.get_firmware_version()
-            if  device_id and firmware_version:
-                # ex Ping1D-id-43-t-1-m-1-v-3.19
-                description = "/dev/serial/ping/Ping1D-id-%s-t-%s-m-%s-v-%s.%s" % (
-                device_id["device_id"],
-                firmware_version["device_type"],
-                firmware_version["device_model"],
-                firmware_version["firmware_version_major"],
-                firmware_version["firmware_version_minor"])
-
-                print("Found Ping1D (ID: %d) at %s" % (device_id["device_id"], dev))
-
-                # Follow link to actual device
-                target_device = subprocess.check_output(' '.join(["readlink", "-f", byiddev]), shell=True)
-                # Strip newline from output
-                target_device = target_device.split('\n')[0]
-
-                # Create another link to it
-                print("Creating symbolic link to", target_device)
-                subprocess.check_output(' '.join(["mkdir", "-p", "/dev/serial/ping"]), shell=True)
-                output = subprocess.check_output("ln -fs %s %s" % (
-                    target_device,
-                    description), shell=True)
-                pings_found += 1
+            # Follow link to actual device
+            target_device = subprocess.check_output(' '.join(["readlink", "-f", "/dev/serial/by-id/%s" % origin]), shell=True)
+            # Strip newline from output
+            target_device = target_device.decode().split('\n')[0]
 
+            # Create another link to it
+            subprocess.check_output(' '.join(["mkdir", "-p", "/dev/serial/ping"]), shell=True)
+            subprocess.check_output("ln -fs %s %s" % (
+                target_device,
+                target), shell=True)
+            print(origin, " linked to ", target)
+            return True
+        except subprocess.CalledProcessError as exception:
+            print(exception)
+            return False
+
+
+    def erase_old_symlinks(self):
+        """
+        Erases all symlinks at "/dev/serial/ping/"
+        """
+        try:
+            subprocess.check_output(["rm", "-rf", "/dev/serial/ping"])
         except subprocess.CalledProcessError as exception:
             print(exception)
-            continue
 
-    exit(pings_found)
+
+    def list_serial_devices(self):
+        """
+        Lists serial devices at "/dev/serial/by-id/"
+        """
+        # Look for connected serial devices
+        try:
+            output = subprocess.check_output("ls /dev/serial/by-id", shell=True)
+            return output.decode().strip().split("\n")
+        except subprocess.CalledProcessError as exception:
+            print(exception)
+            return []
+
+
+    def __init__(self):
```

Comment:
All of this logic should go under main so that PingEnumerator can be imported and used in alternative independent use-cases.

Reply:
(we won't always want to do this when we construct a pingenumerator)